app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,12 @@
 class PostsController < ApplicationController
   def index
-    @user_index = params[:user_id]
+    @user_path = "/users/#{params[:user_id]}"
+    @user = User.find(params[:user_id])
+    @posts = Post.where(author_id: params[:user_id])
   end
 
   def show
-    @user_index = params[:user_id]
-    @post_index = params[:id]
+    @post = Post.find(params[:id])
+    @comments = Comment.where(post_id: params[:id])
   end
```

Comment:
- [ ] <b>REQUIRED:</b> The same thing goes here, you don't have to make another query. to get the comments that belong to that post, you just do `@comments = @post.comments`.  that's the whole point of the relationships you have created on models of post and comments (`post has_many :comments` and `comment belongs_to :post`).  rails make it so that if there is such a relationship between entities, you can be able to get each other on each other's object. like `@comments = @post.comments` when you need comments that belong to a post or `@post = @comment.post` when you need a post to which that comment belongs.